lib/mdl/doc.rb

```diff
@@ -48,8 +48,10 @@ def initialize(text, ignore_front_matter = false)
     def self.new_from_file(filename, ignore_front_matter = false)
       if filename == '-'
         new($stdin.read, ignore_front_matter)
-      else
+      elsif File.exist?(filename)
         new(File.read(filename, :encoding => 'UTF-8'), ignore_front_matter)
+      else
+        raise Errno::ENOENT, filename, []
```

Comment:
raising causes a stracktrace, it's just... a very short. one. :) 
```shell
[phil@rider (lint) RVM(2.7.1) markdownlint]$ ./bin/mdl /tmp/foo
Traceback (most recent call last):
./bin/mdl: No such file or directory - /tmp/foo (Errno::ENOENT)
```

If we wanna be more friendly, it seems like a better approach would be in `lib/mdl.rb` just before `puts "Checking`, do something like:

```ruby
unless File.exist?(filename)
  puts "ERROR: #{filename} not found"
  exit 1
end
```
or something?

Reply:
You're right that does look better. Not sure about putting this in `lib/mdl.rb` though, isn't `new_from_file` a more natural place to put the changes? I'm not really familiar with the project so I might be way off.

Reply:
Sorry I never answered this. I was trying to push this up the stack so that deeper libraries aren't doing output. It's always cleaner to keep the cli (which is effectively what lib/mdl.rb is) doing the output and checking rather than getting deep down into libraries that then throw errors.